config/webpack-metalsmith-connect/index.js

```diff
@@ -27,29 +27,33 @@ function getEntryPoints(buildOptions) {
   return getWebpackEntryPoints(manifestsToBuild);
 }
 
-function connectToMetalsmith(buildOptions, webpackMiddleware) {
-  const convertPathsMiddleware = convertPathsToRelative(buildOptions);
+function compileAssets(buildOptions) {
   return (files, metalsmith, done) => {
+    const convertPathsMiddleware = convertPathsToRelative(buildOptions);
+    const apps = getEntryPoints(buildOptions);
+    const webpackConfig = generateWebpackConfig(buildOptions, apps);
+    const webpackMiddleware = webpackPlugin(webpackConfig);
+
     webpackMiddleware(files, metalsmith, (err) => {
       if (err) throw err;
       convertPathsMiddleware(files, metalsmith, done);
     });
   };
 }
 
-function compileAssets(buildOptions) {
-  const apps = getEntryPoints(buildOptions);
-  const webpackConfig = generateWebpackConfig(buildOptions, apps);
-  const webpackMiddleware = webpackPlugin(webpackConfig);
-  return connectToMetalsmith(buildOptions, webpackMiddleware);
-}
-
 function watchAssets(buildOptions) {
-  const apps = getEntryPoints(buildOptions);
-  const webpackConfig = generateWebpackConfig(buildOptions, apps);
-  const webpackDevServerConfig = generateWebpackDevConfig(buildOptions, manifests);
-  const webpackMiddleware = webpackDevServerPlugin(webpackConfig, webpackDevServerConfig);
-  return connectToMetalsmith(buildOptions, webpackMiddleware);
+  return (files, metalsmith, done) => {
+    const convertPathsMiddleware = convertPathsToRelative(buildOptions);
+    const apps = getEntryPoints(buildOptions);
+    const webpackConfig = generateWebpackConfig(buildOptions, apps);
+    const webpackDevServerConfig = generateWebpackDevConfig(buildOptions);
+    const webpackMiddleware = webpackDevServerPlugin(webpackConfig, webpackDevServerConfig);
+
+    webpackMiddleware(files, metalsmith, (err) => {
+      if (err) throw err;
+      convertPathsMiddleware(files, metalsmith, done);
+    });
+  };
```

Comment:
This file was rearranged so that the configurations are generated inside the middleware function as opposed to when the middleware is registered. The reason for this is that middleware that executes prior to it, `create-build-settings`, will bind the resultant `settings` to `buildOptions`, which can be used by the dev server.